### ISSUE TITLE
jf go-publish [Error] create zip: file "XXX" is larger than declared size

### ISSUE THREAD

**Issue by chkp-roniz** [jfrog/jfrog-cli#2423]
### Describe the bug

Trying to publish go packages fail on several projects to create zip
Debug logs are not helpful in tracing the root cause

I tried to use the latest JF CLI version (2.52.9), and it did not work as well

### Current behavior

```
$ jf gp v1.0.0
08:09:10 [Debug] JFrog CLI version: 2.45.0
08:09:10 [Debug] OS/Arch: linux/amd64
08:09:10 [Debug] Go config file was found in: /root/.jfrog/projects/go.yaml
08:09:10 [Info] Using go: go1.19.13
08:09:10 [Debug] Found deployer in the config file /root/.jfrog/projects/go.yaml
08:09:10 [Debug] Sending HTTP GET request to: https://artifactory.mycompany.com/artifactory/api/system/version
08:09:10 [Debug] Artifactory response: 200 OK
08:09:10 [Debug] JFrog Artifactory version is: x.y.z
08:09:10 [Debug] Saving build general details at: /builds/myproject/build-info/jfrog/builds/7030aee1d311525093fa971fc86f9b46d80112a4fb7a8a175dfdb9ac66732c68/partials
08:09:10 [Info] Running 'go list -mod=mod -m' in /builds/myproject
08:09:10 [Debug] mycompany.com/myproject
08:09:10 [Info] Publishing mycompany.com/myproject to go
08:09:10 [Error] create zip: file "build-info/jfrog.cli.temp.-1704182950-1424564733/project.zip2966673615" is larger than declared size
{
 "status": "failure",
 "totals": {
 "success": 0,
 "failure": 0
 }
}
```

### JFrog CLI version

2.52.9

### Operating system type and version

OS/Arch: linux/amd64

**Comment by chkp-roniz** [comment]
Thanks to the amazing assistance of Vaibhav Jain from JFrog support, we were able to trace the root cause that is related to JFROG_CLI_TEMP_DIR

In our GitLab template (see [more details](https://medium.com/@izra.ron/how-did-we-seamlessly-integrated-jfrog-cli-into-our-gitlab-ci-pipelines-52fd1d6e3119)), we configured JFROG_CLI_TEMP_DIR inside the project directory (the motivation was to share it across jobs while collecting all jobs metadata and publish at the end). It seems like Go Zip has some race conditions when trying to generate a zip file inside a folder that it tries to zip. The issue was only reproduced in some of our go projects.

To resolve it, we simply added an exclusion flag to the command, and the package was successfully published.
```
jf go-publish v0.0.0  --exclusions=$JFROG_CLI_TEMP_DIR/*
```
